cores/src/page_snapshot.rs

```diff
@@ -24,3 +26,14 @@ pub fn clean_raw_html(raw_html: &str, url: &Url) -> color_eyre::Result<String> {
     let article = extractor::extract(&mut raw_html_cursor, url)?;
     Ok(article.content)
 }
+
+pub fn extract_title(content: &str) -> Result<String> {
+    let document = Html::parse_document(content);
+    let title_selector = Selector::parse("title").unwrap();
+
+    if let Some(title_element) = document.select(&title_selector).next() {
+        Ok(title_element.text().collect::<String>())
+    } else {
+        Ok("No title found".to_string())
```

Comment:
This could probably return an `Result<Option<String>>` so that in this case someone who calls this function and react how they want to a title not being found without having to look for this specific error string!